code/modules/power/supermatter/supermatter_hit_procs.dm

```diff
@@ -274,12 +276,14 @@
 				message_admins("[src] has consumed [consumed_object], [suspicion] [ADMIN_JMP(src)].")
 			investigate_log("has consumed [consumed_object] - [suspicion].", INVESTIGATE_ENGINE)
 		qdel(consumed_object)
-	if(!iseffect(consumed_object) && power_changes)
-		matter_power += 200
+	if(!iseffect(consumed_object) && isitem(consumed_object) && power_changes)
+		var/obj/item/consumed_item = consumed_object
+		object_size = consumed_item.w_class
+		matter_power += 70 * object_size
 
 	//Some poor sod got eaten, go ahead and irradiate people nearby.
-	radiation_pulse(src, max_range = 6, threshold = 0.3, chance = 30)
-	for(var/mob/living/near_mob in range(10))
+	radiation_pulse(src, max_range = 1.5 * object_size, threshold = 1.2 / object_size, chance = 10 * object_size)
+	for(var/mob/living/near_mob in range(2 * object_size))
```

Comment:
Actually uh, I get the feeling you might be better off keeping max_range at 6 and range() at 10 here, since radiation I assume has falloff. Otherwise small insertions won't ever be noticed and all, which sucks

Reply:
sure